cli/tests/unit/net_test.ts

```diff
@@ -338,7 +338,7 @@ unitTest(
 unitTest(
   { perms: { net: true } },
   async function netTcpListenIteratorBreakClosesResource(): Promise<void> {
-    const promise = deferred();
+    const promise = new Deferred<void>();
```

Comment:
Is it necessary to put the `<void>` in here?

Reply:
Yes, or else the default type is `{}` and mandates a parameter of type `{}` when resolve is called. This patch was introduced in a later version of typescript though (https://github.com/microsoft/TypeScript/pull/39817)

Reply:
https://github.com/denoland/deno/pull/7573#issue-489666614

Reply:
Writing `new Deferred<void>` is fairly obnoxious. We should reconsider this change.

Reply:
@ry Additionally, extending `Promise` has some annoying behaviours which forces it to [accept an executor function](https://discord.com/channels/684898665143206084/688040863313428503/784110243004940351), which `deferred()` did not.

Reply:
I'm actually pretty happy with current implementation, I think we should only change `Object.assign()` bit to directly assign to `promise`

Reply:
@ry @nayeemrmn, I think we'll need to explicitly write `deferred<void` too. Not mentioning type of `methods` in  https://github.com/denoland/deno/blob/master/std/async/deferred.ts#L21 allows current behaviour but has the downside of not mentioning type. But that can be fixed by adding type to current implementation too.